src/ristretto/ristretto_sig.rs

```diff
@@ -167,11 +165,45 @@ mod test {
     /// Ristretto scalars have a max value 2^255. This test checks that hashed messages above this value can still be
     /// signed as a result of applying modulo arithmetic on the challenge value
     #[test]
+    #[allow(non_snake_case)]
     fn challenge_from_invalid_scalar() {
         let mut rng = rand::thread_rng();
         let m = from_hex("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff").unwrap();
         let k = RistrettoSecretKey::random(&mut rng);
         let r = RistrettoSecretKey::random(&mut rng);
-        assert!(RistrettoSchnorr::sign(k, r, &m).is_ok());
+        assert!(RistrettoSchnorr::sign_raw(&k, r, &m).is_ok());
+    }
+
+    #[test]
+    #[allow(non_snake_case)]
+    fn domain_separated_challenge() {
+        let P =
+            RistrettoPublicKey::from_hex("74896a30c89186b8194e25f8c1382f8d3081c5a182fb8f8a6d34f27fbefbfc70").unwrap();
+        let R =
+            RistrettoPublicKey::from_hex("fa14cb581ce5717248444721242e6b195a482d503a853dea4acb513074d8d803").unwrap();
+        let msg = "Moving Pictures";
+        let hash = SchnorrSignature::construct_domain_separated_challenge::<_, Blake256>(&R, &P, msg);
+        let naiive = Blake256::new()
+            .chain(R.as_bytes())
+            .chain(P.as_bytes())
```

Comment:
If the other comment about swapping these two challenge inputs is accepted, there should be a corresponding swap here.